FabricObserver/Observers/AppObserver.cs

```diff
@@ -2057,13 +2189,65 @@ private Task<ParallelLoopResult> MonitorDeployedAppsAsync(CancellationToken toke
 
                 // Memory \\
 
+                // Private Bytes.
+                if (checkMemPrivateBytes)
+                {
+                    float memPb = ProcessInfoProvider.Instance.GetProcessPrivateBytesMb(procId);
+                    
+                    if (procId == parentPid)
+                    {
+                        AllAppPrivateBytesDataMb[id].AddData(memPb);
+
+                        // RG
+                        if (rgMemoryPercentThreshold > 0 && repOrInst.RGMemoryLimitMb > 0)
+                        {
+                            double pct = ((double)memPb / repOrInst.RGMemoryLimitMb) * 100;
+                            AllAppRGMemoryUsagePercent[id].AddData(pct);
+                        }
+                    }
+                    else
+                    {
+                        _ = AllAppPrivateBytesDataMb.TryAdd(
+                               $"{id}:{procName}{procId}",
+                               new FabricResourceUsageData<float>(
+                                       ErrorWarningProperty.PrivateBytesMb,
+                                       $"{id}:{procName}{procId}",
+                                       capacity,
+                                       UseCircularBuffer,
+                                       EnableConcurrentMonitoring));
+                        
+                        AllAppPrivateBytesDataMb[$"{id}:{procName}{procId}"].AddData(memPb);
+                        
+                        // RG
+                        if (rgMemoryPercentThreshold > 0 && repOrInst.RGMemoryLimitMb > 0)
+                        {
+                            _ = AllAppRGMemoryUsagePercent.TryAdd(
+                                $"{id}:{procName}{procId}",
+                                new FabricResourceUsageData<double>(
+                                        ErrorWarningProperty.RGMemoryUsagePercent,
+                                        $"{id}:{procName}{procId}",
+                                        capacity,
+                                        UseCircularBuffer,
+                                        EnableConcurrentMonitoring));
+                            
+                            double pct = ((double)memPb / repOrInst.RGMemoryLimitMb) * 100;
+                            AllAppRGMemoryUsagePercent[$"{id}:{procName}{procId}"].AddData(pct);
+                        }
+                    }
+                }
+
+                // Working Set.
                 if (checkMemMb)
                 {
                     if (procId == parentPid)
                     {
-                        _ = ProcessInfoProvider.Instance.GetProcessWorkingSetMb(procId, procName, Token, checkPrivateWS);
-                        Thread.Sleep(100);
-                        AllAppMemDataMb[id].AddData(ProcessInfoProvider.Instance.GetProcessWorkingSetMb(procId, procName, Token, checkPrivateWS));
+                        if (usePerfCounter)
```

Comment:
do we need to use "usePerfCounter" while checking private bytes above as well?

Reply:
No. The current impl is less CPU intensive than spinning up a PerfCounter for Process/Private Bytes. The only reason that a perf counter is used for Process/Working Set - Private is because there is no (publicly documented) way to get Private Working Set size from a Win32 API call (I know how to do this, but can't (the related structure is not documented).